src/tools/rpc.rs

```diff
@@ -0,0 +1,224 @@
+}
+
+/// [EncodedBlockCert] defines how get-block response encodes a block and its certificate.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct EncodedBlockCert {
+    /// Block header data.
+    pub block: BlockHeaderMsgPack,
+    /// Certificate.
+    pub cert: Certificate,
+}
+
+/// A Certificate contains a cryptographic proof that agreement was reached on a
+/// given block in a given round.
+///
+/// When a client first joins the network or has fallen behind and needs to catch
+/// up, certificates allow the client to verify that a block someone gives them
+/// is the real one.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Certificate {
+    /// Proposal value.
+    #[serde(default)]
+    pub prop: Option<CertificateProposal>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CertificateProposal {
+    /// Block header's hash.
+    #[serde(rename = "dig")]
+    pub block_digest: HashDigest,
+}
+
+/// BlockHeader
+/// Deserialized from MessagePack format.
+///
+/// See [block.go](https://github.com/algorand/go-algorand/blob/master/data/bookkeeping/block.go) for more details.
+// Comments below are simply copied from the go-algorand repo.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BlockHeaderMsgPack {
+    /// RewardsLevel specifies how many rewards, in MicroAlgos, have been distributed
+    /// to each config.Protocol.RewardUnit of MicroAlgos since genesis.
+    #[serde(default)]
+    pub earn: u64,
+
+    /// The FeeSink accepts transaction fees. It can only spend to the incentive pool.
+    #[serde(default)]
+    pub fees: Option<HashDigest>,
+
+    /// The number of leftover MicroAlgos after the distribution of RewardsRate/rewardUnits
+    /// MicroAlgos for every reward unit in the next round.
+    #[serde(default)]
+    pub frac: u64,
+
+    /// Genesis ID to which this block belongs.
+    #[serde(default)]
+    pub gen: String,
+
+    /// Genesis hash to which this block belongs.
+    #[serde(default)]
+    pub gh: Option<HashDigest>,
+
+    /// The hash of the previous block.
+    #[serde(default)]
+    pub prev: Option<HashDigest>,
+
+    /// Current protocol.
+    #[serde(default)]
+    pub proto: String,
+
+    /// The number of new MicroAlgos added to the participation stake from rewards at the next round.
+    #[serde(default)]
+    pub rate: u64,
+
+    /// Round represents a protocol round index.
+    #[serde(default)]
+    pub rnd: u64,
+
+    /// The round at which the RewardsRate will be recalculated.
+    #[serde(default)]
+    pub rwcalr: u64,
+
+    /// The RewardsPool accepts periodic injections from the FeeSink and continually
+    /// redistributes them to addresses as rewards.
+    #[serde(default)]
+    pub rwd: Option<HashDigest>,
+
+    /// Sortition seed.
+    #[serde(default, deserialize_with = "deserialize_byte32_arr_opt")]
+    pub seed: Option<[u8; 32]>,
+
+    /// TimeStamp in seconds since epoch.
+    #[serde(default)]
+    pub ts: i64,
```

Comment:
Maybe make names better and rename them with serde?

Reply:
Done